neo/Ledger/Blockchain.cs

```diff
@@ -262,7 +273,7 @@ private RelayResultReason OnNewBlock(Block block)
                 return RelayResultReason.AlreadyExists;
             if (block.Index - 1 >= header_index.Count)
             {
-                block_cache_unverified[block.Index] = block;
+                AddUnverifiedBlockToCache(block);
                 return RelayResultReason.UnableToVerify;
```

Comment:
Maybe we could check here if the block was signed by the required number of CN nodes, then perhaps we wouldn’t need to keep a linked list of the blocks, since it shouldn’t be possible to have the CN nodes sign off on two blocks at the same height.

Reply:
Seems reasonable. I’m going to push this it seems to be running well.